lib/puma/cluster.rb

```diff
@@ -322,6 +343,34 @@ def setup_signals
               w.last_status[:requests_count] >= fork_requests
           end
         end
+
+      end
+      if @options[:mold_worker]
+        Signal.trap "SIGURG" do
+          mold_and_refork!
+        end
+        current_interval_index = 0
+        Signal.trap "SIGUSR1" do
+          # stop any existing mold
+          @mold&.term
+          # begin a full restart
+          phased_restart
+          # reset the mold_worker intervals
+          current_interval_index = 0
+          wakeup!
+        end
+        @events.register(:ping!) do |w|
+          # if there's a phased_restart under way don't step on its toes
+          next unless all_workers_in_phase?
+
+          next_request_interval = @options[:mold_worker][current_interval_index]
+          next unless next_request_interval && w.last_status[:requests_count] >= next_request_interval
+          # if @phase has already been incremented, we're already reforking;
```

Comment:
Is multiple interval case tested in spec?

Reply:
It is not; the specs require CLI arguments and I thought rather than make the CLI argument complex (and allow comma-delimited arrays like `--mold_worker=200,400,800`) I would just keep the CLI functionality minimal and keep the extended feature to the config DSL.